### ISSUE TITLE
no data in the pinpoint web. we use nginx and php 7.1.18. dose the nginx server is supported by this agent?

### ISSUE THREAD

**Issue by nicedu123** [naver/pinpoint-c-agent#83]
2019-06-21 12:26:27 [pinpoint] [19812] pinpoint_api.cpp:193 [INFO] common.CollectorSpanPort=9996
 2019-06-21 12:26:27 [pinpoint] [19812] pinpoint_api.cpp:193 [INFO] common.CollectorStatPort=9995
 2019-06-21 12:26:27 [pinpoint] [19812] pinpoint_api.cpp:193 [INFO] common.CollectorTcpPort=9994
 2019-06-21 12:26:27 [pinpoint] [19812] pinpoint_api.cpp:169 [INFO]  [test.ApiTableFile] config not find use default [NULL]
 2019-06-21 12:26:42 [pinpoint] [19814] trace_data_sender.cpp:42 [INFO] TraceDataSender::init() start. 
 2019-06-21 12:26:42 [pinpoint] [19814] api_data_sender.cpp:240 [INFO] ApiDataSender::init() start. 
 2019-06-21 12:26:42 [pinpoint] [19814] string_data_sender.cpp:95 [INFO] StringDataSender::init() start. 
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_agent.cpp:471 [INFO] PinpointAgent::registerPlugins len(plugins)=4
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_agent.cpp:472 [INFO] PinpointAgent::registerPlugins include=PhpRequestPlugin;QuickStartPlugin  ; PhpInternalFunctionsPlugin ;ExcludePlugin ;PDOPlugin;CurlPlugin;
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_agent.cpp:473 [INFO] PinpointAgent::registerPlugins exclude=ExcludePlugin
 2019-06-21 12:26:42 [pinpoint] [19814] request_plugin.cpp:90 [INFO] PhpRequestInterceptor::init() SUCCESS!
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_agent.cpp:514 [INFO] plugin=ExcludePlugin in excludes, ignore!
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_ext.cpp:411 [INFO]  pinpoint agent start !!! code = 0
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_client.cpp:193 [INFO] start_connect: 172.18.97.33:9994
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_client.h:272 [INFO] state change: 1 => 3
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_client.cpp:251 [INFO] 172.18.97.33:9994 Connect success.
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_client.h:272 [INFO] state change: 3 => 10
 2019-06-21 12:26:42 [pinpoint] [19814] agent_data_sender.cpp:123 [INFO] TAgentInfo: [0x7f7ad80134d0]
 2019-06-21 12:26:42 [pinpoint] [19815] trace_data_sender.cpp:42 [INFO] TraceDataSender::init() start. 
 2019-06-21 12:26:42 [pinpoint] [19815] api_data_sender.cpp:240 [INFO] ApiDataSender::init() start. 
 2019-06-21 12:26:42 [pinpoint] [19815] string_data_sender.cpp:95 [INFO] StringDataSender::init() start. 
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_agent.cpp:471 [INFO] PinpointAgent::registerPlugins len(plugins)=4
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_agent.cpp:472 [INFO] PinpointAgent::registerPlugins include=PhpRequestPlugin;QuickStartPlugin  ; PhpInternalFunctionsPlugin ;ExcludePlugin ;PDOPlugin;CurlPlugin;
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_agent.cpp:473 [INFO] PinpointAgent::registerPlugins exclude=ExcludePlugin
 2019-06-21 12:26:42 [pinpoint] [19815] request_plugin.cpp:90 [INFO] PhpRequestInterceptor::init() SUCCESS!
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_agent.cpp:514 [INFO] plugin=ExcludePlugin in excludes, ignore!
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_client.cpp:193 [INFO] start_connect: 172.18.97.33:9994
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_ext.cpp:411 [INFO]  pinpoint agent start !!! code = 0
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_client.h:272 [INFO] state change: 1 => 3
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_client.cpp:251 [INFO] 172.18.97.33:9994 Connect success.
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_client.h:272 [INFO] state change: 3 => 10
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(0)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:234 [INFO] handshake response code = Duplex Connection successfully established.
 2019-06-21 12:26:42 [pinpoint] [19814] pinpoint_client.h:272 [INFO] state change: 10 => 12
 2019-06-21 12:26:42 [pinpoint] [19815] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(0)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:42 [pinpoint] [19815] hand_shake_sender.cpp:234 [INFO] handshake response code = Duplex Connection successfully established.
 2019-06-21 12:26:42 [pinpoint] [19815] pinpoint_client.h:272 [INFO] state change: 10 => 12
 2019-06-21 12:26:42 [pinpoint] [19815] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(1)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:42 [pinpoint] [19815] hand_shake_sender.cpp:234 [INFO] handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:42 [pinpoint] [19815] hand_shake_sender.cpp:239 [INFO] old handshake response code = Duplex Connection successfully established.
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(1)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:234 [INFO] handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:239 [INFO] old handshake response code = Duplex Connection successfully established.
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(1)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:234 [INFO] handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:239 [INFO] old handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(1)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:234 [INFO] handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:42 [pinpoint] [19814] hand_shake_sender.cpp:239 [INFO] old handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:43 [pinpoint] [19814] pinpoint_ping.cpp:63 [INFO] send ping: pingId=0, stateVersion=0, stateCode=12
 2019-06-21 12:26:43 [pinpoint] [19814] hand_shake_sender.cpp:205 [INFO] get control message: ControlMessage[type=map; data=(ControlMessage[type=string; data=(code)]=>ControlMessage[type=long; data=(1)], ControlMessage[type=string; data=(subCode)]=>ControlMessage[type=long; data=(2)])]
 2019-06-21 12:26:43 [pinpoint] [19814] hand_shake_sender.cpp:234 [INFO] handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:43 [pinpoint] [19814] hand_shake_sender.cpp:239 [INFO] old handshake response code = Already Duplex Connection established.
 2019-06-21 12:26:43 [pinpoint] [19815] pinpoint_ping.cpp:63 [INFO] send ping: pingId=0, stateVersion=0, stateCode=12
 2019-06-21 12:31:43 [pinpoint] [19814] pinpoint_ping.cpp:63 [INFO] send ping: pingId=1, stateVersion=0, stateCode=12
 2019-06-21 12:31:43 [pinpoint] [19815] pinpoint_ping.cpp:63 [INFO] send ping: pingId=1, stateVersion=0, stateCode=12
 2019-06-21 12:36:43 [pinpoint] [19814] pinpoint_ping.cpp:63 [INFO] send ping: pingId=2, stateVersion=0, stateCode=12
 2019-06-21 12:36:43 [pinpoint] [19815] pinpoint_ping.cpp:63 [INFO] send ping: pingId=2, stateVersion=0, stateCode=12
 2019-06-21 12:41:43 [pinpoint] [19814] pinpoint_ping.cpp:63 [INFO] send ping: pingId=3, stateVersion=0, stateCode=12
 2019-06-21 12:41:43 [pinpoint] [19815] pinpoint_ping.cpp:63 [INFO] send ping: pingId=3, stateVersion=0, stateCode=12

this is the pinpoint_log.text

**Comment by EyelynSu** [comment]
Nginx is supported.
What function have you called? And have you added plugin for the function?
You can see the sample of plugin in https://github.com/naver/pinpoint-c-agent/tree/dev/simulate/web/php/plugins and API https://github.com/naver/pinpoint-c-agent/wiki/Pinpoint-Agent-PHP-API.
Your pinpoint_log.txt's log level is INFO, you can set it to DEBUG in pinpoint_agent.conf , then you can
get more detailed infomation.

**Comment by nicedu123** [comment]
Hi ,@EyelynSu
i have change the log level to DEBUG. Here are some of logs:
  2019-06-24 10:33:16 [pinpoint] [29528] pinpoint_ping.cpp:63 [INFO] send ping: pingId=0, stateVersion=0, stateCode=12
 2019-06-24 10:33:16 [pinpoint] [29528] executor.cpp:73 [DEBUG] ScheduledExecutor: scheduledExecutor add task: name=pingTask, interval=300000, callTimes=1
 2019-06-24 10:33:16 [pinpoint] [29528] executor.cpp:94 [DEBUG] run addTask_ in io.run() thread: add pingTask
 2019-06-24 10:33:16 [pinpoint] [29528] pinpoint_client.cpp:540 [DEBUG] get Pong
 2019-06-24 10:33:45 [pinpoint] [29528] agent_monitor_sender.cpp:128 [TRACE] TAgentStatBatch: [TAgentStatBatch(agentId=php_pinpoint, startTimestamp=-2024512404, agentStats=[TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>)])]
 [root@devops8 logs]# tail -f pinpoint_log.txt 
 2019-06-24 10:36:36 [pinpoint] [29528] pinpoint_agent.cpp:49 [DEBUG] beyondOneSecond: start=2019-Jun-24 02:36:36
 2019-06-24 10:36:36 [pinpoint] [29528] pinpoint_agent.cpp:50 [DEBUG] beyondOneSecond: end=2019-Jun-24 02:36:36
 2019-06-24 10:36:36 [pinpoint] [29528] pinpoint_agent.cpp:52 [DEBUG] beyondOneSecond: second=0
 2019-06-24 10:36:36 [pinpoint] [29528] memory_pool.h:118 [DEBUG] SafeShareMemory ProcsSharefeData-1561343583340 size 0
 2019-06-24 10:36:36 [pinpoint] [29528] memory_pool.h:125 [DEBUG] ProcsSharefeData-1561343583340 expand 208  createrPid 29528 
 2019-06-24 10:36:36 [pinpoint] [29528] memory_pool.h:135 [DEBUG] SafeShareMemory:ProcsSharefeData-1561343583340 pid:29528 
 2019-06-24 10:36:36 [pinpoint] [29528] trace_data_sender.cpp:88 [TRACE] Span: [TSpan(agentId=php_pinpoint, applicationName=devops8_new, agentStartTime=1561343583340, transactionId=php_pinpoint^1561343583340^2, spanId=2494802244809351178, parentSpanId=-1, startTime=1561343796838, elapsed=1, rpc=/test.php, serviceType=1500, endPoint=47.106.217.76, remoteAddr=193.42.40.108, annotations=[TAnnotation(key=46, value=TAnnotationValue(stringValue=<null>, boolValue=<null>, intValue=200, longValue=<null>, shortValue=<null>, doubleValue=<null>, binaryValue=<null>, byteValue=<null>, intStringValue=<null>, intStringStringValue=<null>, longIntIntByteByteStringValue=<null>, intBooleanIntBooleanValue=<null>))], flag=0, err=<null>, spanEventList=[], parentApplicationName=<null>, parentApplicationType=<null>, acceptorHost=<null>, apiId=-11, exceptionInfo=<null>, applicationServiceType=<null>, loggingTransactionInfo=<null>)]
 2019-06-24 10:36:36 [pinpoint] [29528] interceptor.cpp:136 [DEBUG] reqShutdownClean called
 2019-06-24 10:36:45 [pinpoint] [29528] agent_monitor_sender.cpp:128 [TRACE] TAgentStatBatch: [TAgentStatBatch(agentId=php_pinpoint, startTimestamp=-2024512404, agentStats=[TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>)])]
 2019-06-24 10:37:15 [pinpoint] [29528] agent_monitor_sender.cpp:128 [TRACE] TAgentStatBatch: [TAgentStatBatch(agentId=php_pinpoint, startTimestamp=-2024512404, agentStats=[TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>)])]
 2019-06-24 10:37:45 [pinpoint] [29528] agent_monitor_sender.cpp:128 [TRACE] TAgentStatBatch: [TAgentStatBatch(agentId=php_pinpoint, startTimestamp=-2024512404, agentStats=[TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>), TAgentStat(agentId=php_pinpoint, startTimestamp=1561343583340, timestamp=<null>, collectInterval=<null>, gc=<null>, cpuLoad=TCpuLoad(jvmCpuLoad=<null>, systemCpuLoad=0), transaction=<null>, activeTrace=<null>, metadata=<null>)])]

**Comment by EyelynSu** [comment]
Would you please give us more infomations, such as the function you called, the plugins you include, your pinpoint_agent.conf, the whole pinpoint_log.txt(from the agent start). etc.

**Comment by nicedu123** [comment]
> Would you please give us more infomations, such as the function you called, the plugins you include, your pinpoint_agent.conf, the whole pinpoint_log.txt(from the agent start). etc.
here are the config and logs. i use the demo in the php/web to test
[pinpoint_log.txt](https://github.com/naver/pinpoint-c-agent/files/3319448/pinpoint_log.txt)

[pinpoint_agent.conf.txt](https://github.com/naver/pinpoint-c-agent/files/3319450/pinpoint_agent.conf.txt)

**Comment by ChenGXQQ** [comment]
This log means that the trace information has been sent to pinpoint collector.
```
2019-06-24 10:43:26 [pinpoint] [29528] trace_data_sender.cpp:88 [TRACE] Span: [TSpan(agentId=php_pinpoint, applicationName=devops8_new, agentStartTime=1561343583340, transactionId=php_pinpoint^1561343583340^4, spanId=3733090354283967896, parentSpanId=-1, startTime=1561344206781, elapsed=0, rpc=/test.php, serviceType=1500, endPoint=, remoteAddr=**, annotations=[TAnnotation(key=46, value=TAnnotationValue(stringValue=, boolValue=, intValue=200, longValue=, shortValue=, doubleValue=, binaryValue=, byteValue=, intStringValue=, intStringStringValue=, longIntIntByteByteStringValue=, intBooleanIntBooleanValue=))], flag=0, err=, spanEventList=[], parentApplicationName=, parentApplicationType=, acceptorHost=, apiId=-11, exceptionInfo=, applicationServiceType=, loggingTransactionInfo=)]
```
@nicedu123 Can you check your pinpoint collector's log? See if the packet is received correctly.

**Comment by eeliu** [comment]
hi, pinpoint-c-agent released v0.2.0,we recommend you upgrade to v0.2.0. It provides a more easy and stable way to trace and debug. 
So, I close this issue.
